### ISSUE TITLE
Color issues in 20220408

### ISSUE THREAD

**Issue by jpalus** [neomutt/neomutt#3349]
Moving color related issues from #3343. Possibly might be related to #3348 but can't really tell:
1. Indicator color appears to be overridden by `mono` entry that should not apply
   - config with just mbox and `color indicator color234 color136` in my terminal:
     ![image](https://user-images.githubusercontent.com/484155/162639748-5e691653-750f-4270-b132-08523dc488a9.png)
   - adding entry `mono indicator reverse` that I believe should be ignored results in:
     ![image](https://user-images.githubusercontent.com/484155/162639821-362b97d1-170a-4c4b-a9b4-178b4115187d.png)
2. Background at the end of quoted text appears to be wrong. Minimal config with just mbox and:
   ```
   color normal     color33        color234
   color hdrdefault     color33        color234
   color quoted     color33        color234
   color message     color33        color234
   ```
   gives:
   ![image](https://user-images.githubusercontent.com/484155/162639959-7ea36cf8-a3e0-48f8-b894-25b3ec43d213.png)
3. Not sure whether this is an actual issue but behavior changed compared to previous release. mailbox that has new mail has a special color in a sidebar. After going to this mailbox, without reading new mail and going back to previous mailbox:
    - neomutt 20211029 used to reset colors to the same as those mailboxes without new mail
    - neomutt 20220408 started using some third fg/bg, different from those for mailbox with and without new mail
      * colors with new mail before visiting:
        ![image](https://user-images.githubusercontent.com/484155/162640496-64cfc922-3d30-43ed-a685-5fd30bda2575.png)
      * colors with new mail after visiting:
        ![image](https://user-images.githubusercontent.com/484155/162640506-acf20b0f-3383-4bf3-b043-94f462df8f68.png)
      * colors without any email:
        ![image](https://user-images.githubusercontent.com/484155/162640415-e020320b-b056-46ed-b144-bee50732ff20.png)

**Comment by jpalus** [comment]
@flatcap confirmed that current `devel/colour` fixes second issue. The other two sill present.

Regarding third issue, sidebar related. I believe relevant color is `sidebar_unread`, two notes regarding it:
* `neomuttrc.5` does not mention this color
* manual states that if not configured explicitly `default` color should be used. What `default` means here exactly? Cause it appears to mean different thing for `sidebar_ordinary` and `sidebar_unread` -- I have customized none of them yet they have different colors.

**Comment by flatcap** [comment]
Thanks.  I'm working on the others.

> `neomuttrc.5` does not mention this color

The manual pages are a pain to update, so they're seldom up-to-date.

> What `default` means here exactly?

Without any config, `default` will be the terminal's background colour.
In many terminals this colour can be transparent (showing the desktop through).

If you configure `color normal red green` then most text will become this colour.
As I'm discovering, the coverage isn't consistent.

> it appears to mean different thing for `sidebar_ordinary` and `sidebar_unread`

I'll check

Thanks for your help and your patience :-)

**Comment by flatcap** [comment]
Sorry for the delay, I've been working through all 46 colours on all screens to make sure they blend correctly with the "normal" colour.
I _think_ I've got it!

1. `mono` is dead.  The code is a no-op.

2. Quoted text should be coloured correctly.
   The text will be coloured according to the normal and quoted1..9 colours.
   The rest of the line will be normal coloured.

3. The sidebar colours now merge in priority order:
    new, unread, flagged, spool_file, ordinary (then overlaid with highlight or indicator)

New: unread, unseen
Old: unread, but seen (folder has been opened)
Ordinary: folder has no exciting characteristics (this is useful for dimming unimportant folders)

Please can you give `master` (e13d420cd) a try.

**Comment by jpalus** [comment]
Checked current master (be16c31) and all 3 issues are resolved. Thanks!

**Comment by cpence** [comment]
Reporting in that my personal color scheme was also broken and is totally resolved on current master (db88c3e at time of posting). Thanks for the fix!
